pydantic/_internal/_dataclasses.py

```diff
@@ -32,19 +31,28 @@ def __init__(self, *args: object, **kwargs: object) -> None:
             pass
 
     class PydanticDataclass(StandardDataclass, typing.Protocol):
-        __pydantic_validator__: typing.ClassVar[SchemaValidator]
         __pydantic_core_schema__: typing.ClassVar[core_schema.CoreSchema]
+        __pydantic_validator__: typing.ClassVar[SchemaValidator]
         __pydantic_serializer__: typing.ClassVar[SchemaSerializer]
         __pydantic_decorators__: typing.ClassVar[_decorators.DecoratorInfos]
         """metadata for `@validator`, `@root_validator` and `@serializer` decorators"""
         __pydantic_fields__: typing.ClassVar[dict[str, FieldInfo]]
         __pydantic_config__: typing.ClassVar[ConfigDict]
 
 
-def prepare_dataclass(
+def set_dataclass_fields(cls: type[StandardDataclass], types_namespace: dict[str, Any] | None = None) -> None:
```

Comment:
I split `prepare_dataclass` into this function and `complete_dataclass` to parallel the `set_model_fields` and `complete_model` functions from BaseModel.

More work is necessary to unify these things as much as we possibly can, but I think it's best to go one step at a time. I'm trying to move toward unifying things to make it easier to tell if/where we can merge the code paths.